DNN Platform/Components/ClientDependency/Source/DependencySorter.cs

```diff
@@ -10,6 +10,30 @@ namespace ClientDependency.Core
     /// </summary>
     internal static class DependencySorter
     {
+
+        public static IList<IClientDependencyFile> FilterDependencies(IList<IClientDependencyFile> dependencies)
+        {
+            if (dependencies.Any(f => f.Name != ""))
+            {
+                var newList = dependencies.Where(f => f.Name == "").ToList();
+                var frameworks = dependencies.Where(f => f.Name != "").GroupBy(f => f.Name.ToLower());
+                foreach (var framework in frameworks)
+                {
+                    var topPriority = framework.FirstOrDefault(d => d.ForceVersion);
```

Comment:
It's not that big of a deal, and we should definitely make sure that the
documentation says that it's only for skin developers.  However, making it
do the (as much as possible) right thing in that scenario isn't difficult,
so why not?


On Wed, Nov 25, 2015 at 3:01 PM, Brian Dukes <bdukes@engagesoftware.com>
wrote:

> ​I use the various JS & CSS controls in my module all the time.  I prefer
> to keep my views as declarative as possible.​
>
>
> On Wed, Nov 25, 2015 at 2:38 PM, Peter Donker <notifications@github.com>
> wrote:
>
>> In DNN Platform/Components/ClientDependency/Source/DependencySorter.cs
>> <https://github.com/dnnsoftware/Dnn.Platform/pull/1028#discussion_r45916639>
>> :
>>
>> > @@ -10,6 +10,30 @@ namespace ClientDependency.Core
>> >      /// </summary>
>> >      internal static class DependencySorter
>> >      {
>> > +
>> > +        public static IList<IClientDependencyFile> FilterDependencies(IList<IClientDependencyFile> dependencies)
>> > +        {
>> > +            if (dependencies.Any(f => f.Name != ""))
>> > +            {
>> > +                var newList = dependencies.Where(f => f.Name == "").ToList();
>> > +                var frameworks = dependencies.Where(f => f.Name != "").GroupBy(f => f.Name.ToLower());
>> > +                foreach (var framework in frameworks)
>> > +                {
>> > +                    var topPriority = framework.FirstOrDefault(d => d.ForceVersion);
>>
>> This property is not available through the Client Resource Manager
>> methods, but only in the control. This effectively makes it only available
>> to skin developers. So you're down to one person.
>>
>> If someone decides to "hack" or abuse the feature, the result is
>> unpredictable. But at the end of the day it's down to instruction: this is
>> only meant for Skin designers, not the modules.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/dnnsoftware/Dnn.Platform/pull/1028/files#r45916639>.
>>
>
>